src/pip/_internal/resolution/resolvelib/provider.py

```diff
@@ -257,11 +257,17 @@ def _eligible_for_upgrade(identifier: str) -> bool:
             is_satisfied_by=self.is_satisfied_by,
         )
 
-    @lru_cache(maxsize=None)
     def is_satisfied_by(self, requirement: Requirement, candidate: Candidate) -> bool:
-        return requirement.is_satisfied_by(candidate)
+        return _pip_provider_is_satisfied_by(requirement, candidate)
```

Comment:
Would a staticmethod work instead? I'm not a huge fan of breaking this out into a helper function

Reply:
Agreed, and I used this hacky code to assure myself this works as expected:

```python
class C:
    def __init__(self): ...

    @staticmethod
    @lru_cache
    def compute(): return 1

    def __del__(self):
        print("C destructor called")

c = C()
c.compute()
del c
print("Deleted c")
```

The destructor call should print before the "Deleted c" is printed, which does not happen if `compute` is a regular method and `lru_cache` is used.